app/src/main/java/com/garethevans/church/opensongtablet/OptionMenuListeners.java

```diff
@@ -2629,9 +2630,13 @@ public void onClick(View v) {
         autoscrollActivatedSwitch.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
             @Override
             public void onCheckedChanged(CompoundButton compoundButton, boolean b) {
-                StaticVariables.clickedOnAutoScrollStart = b;
-                if (!b && mListener!=null) {
-                    mListener.stopAutoScroll();
+                // gesture contains the start/stop logic.
+                if (mListener != null) {
+                    // Request a start/stop to get to the desired state
+                    if ((!StaticVariables.isautoscrolling && b) || (StaticVariables.isautoscrolling && !b)) {
```

Comment:
Maybe change to
if (StaticVariables.isautoscrolling!=b) {

Reply:
Sure, I kinda coded my thoughts a little too much!  I think there is another case of of this 'over thinking'.

Reply:
Glad I'm not the only one!